libmscore/barline.cpp

```diff
@@ -805,6 +823,18 @@ Element* BarLine::drop(EditData& data)
             score()->undoAddElement(atr);
             return atr;
             }
+      else if (e->isSymbol()) {
+            e->setParent(this);
```

Comment:
Removing strings 827-831 and adding `|| e->isImage()` to the `if` statement will help to avoid code duplication.

Reply:
And otherwise, we might want to apply the same rules to the future descendants implicitly and don't waste our time for debugging all possible usage of similar classes. I like the idea of using `isBSymbol()` here if we are sure that the code below is applied to the semantic of `BSymbol` class, not that two descendants only.